### ISSUE TITLE
Windows on secondary monitor placed outside of screen when pressing Super-D ("Show workspaces") 

### ISSUE THREAD

**Issue by ullgren** [pop-os/gnome-shell#102]
**(1) Issue/Bug Description:**

When pressing Super-D to "Show Workspaces" the windows on the secondary monitor is places slightly off screen.

This started after upgrading to 22.04

**(2) Steps to reproduce (if you know):**

* Two monitors
* Configure to only have workspaces on primary monitor
* Open windows and place some of them on secondary monitor
* Press Super-D

**(3) Expected behavior:**

The windows on the secondary monitor should show as on the primary.

If switching to use "Workspaces Span Displays" the windows show as expected.

**(4) Distribution (run `cat /etc/os-release`):**
 
```
ponu@binky:~$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

**(5) Gnome Shell version:**
42

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**

```
ponu@binky:~$ apt policy pop-shell
pop-shell:
  Installed: 1.1.0~1649099191~22.04~cb76a4c
  Candidate: 1.1.0~1649099191~22.04~cb76a4c
  Version table:
 *** 1.1.0~1649099191~22.04~cb76a4c 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
       1001 http://apt.pop-os.org/release jammy/main i386 Packages
        100 /var/lib/dpkg/status

```

**(7) Where was Pop Shell installed from:**

Using pop_os! repos

Just upgraded from 21.10

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**

Primary : 1920x1080
Secondary: 1680x1050

**(9) Other Installed/Enabled Extensions:**

**(10) Other Notes:**

If it is of interest I have a screen cast showing the behavior.

**Comment by RagingLoon** [comment]
Same issue when set to "Workspaces on Primary Display Only" via Desktop Options in settings....

Here's video of it https://youtu.be/bU8DQOyi4y4

![mpv-shot0231](https://user-images.githubusercontent.com/57045751/166307769-e99ae215-f664-4cc9-a4ad-d91e6485c695.jpg)

![image](https://user-images.githubusercontent.com/57045751/166299708-0d7e7417-48ac-4efd-9c70-f74088a74088.png)

**Comment by Michael-Duivestein** [comment]
Same issue with two monitors in vertical configuration (1920x2160)
Further, when in workspace overview and dragging a window from the primary monitor to the secondary monitor, the drop zone is limited to the far left side of the screen.
![workspaces-normal](https://user-images.githubusercontent.com/20239555/166450461-2a212e9c-80a3-4f37-8e71-86bb8739d86d.png)
![workspaces-overview](https://user-images.githubusercontent.com/20239555/166450463-e5aaeca5-5734-4eed-ac59-2fca0658c649.png)


Edit: I'm fairly certain this same problem occurred a few releases ago, then was fixed. Can't remember exactly when.

**Comment by ahoneybun** [comment]
I'm seeing this issue as well, I refreshed the OS and the issue was still there. I just tested a different account and the issue isn't there so it is something config related.

**Comment by ullgren** [comment]
@ahoneybun findings made me create a new user and then copy over my old users .config file by file.
Turns out the setting that triggers this behavior is the "Placement of the Workspace Picker" when "Along the left side" (default) the windows on the second monitor are shown as expected. Selecting "Along the right side" triggers the bug.

Workaround is to live with the default setting for this setting (show below)
![workspaces-to-the-left](https://user-images.githubusercontent.com/536239/175810667-904d0392-4746-4fd6-b95f-c3b83d05cab5.png)

**Comment by ahoneybun** [comment]
Seems to be the same for me as well, using either "Along the left side" or "Workspaces Span Displays" work without issue but using "Along the right side" and "Workspaces on Primary Display Only" causes the issue.

**Comment by Michael-Duivestein** [comment]
Last night I upgraded my other machine. Before upgrading I changed "Placement of the Workspace Picker" to "Along the left side".

Post-upgrade, workspaces on the secondary monitor worked normally. Changing "Placement of the Workspace Picker" to "Along the right side" resulted in the workspace problem in this issue.

This would indicate that the placement of the picker pre-upgrade is not linked to this issue.

**Comment by ullgren** [comment]
@Michael-Duivestein No agree that the problem is not connected to if the change is made before or after the upgrade. Sorry if my previous comment was unclear. 
To me it seems that there is a bug on how the offset is calculated when the Workspace Picker is on the right side of the screen.

**Comment by Michael-Duivestein** [comment]
@ullgren I completely agree with you. Your comment simply gave me a theory, which I decided to test. The results help to define the bug's scope, which will aid whoever is assigned to it.

**Comment by RagingLoon** [comment]
Same here. I've been using workspaces span displays since I first noticed this bug and I've absolutely hated it (tried to like it) but at least everything didn't get shoved off the screen. 

Thanks for the tip I just switched to along the left side and everything is fine.

**Comment by danilopinotti** [comment]
Same here. The only solution that worked until now was changing to show the workspaces along the left side.
